.github/workflows/build.yml

```diff
@@ -0,0 +1,38 @@
+name: build
+
+on:
+  push:
+    branches:
+    - master
+
+jobs:
+  # Build on multiple platforms for continuous integration
+  build:
+    strategy:
+      matrix:
+        os: [ ubuntu-latest, macos-13, ubuntu-24.04-arm, macos-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v4
+    - uses: cachix/install-nix-action@v31
+      with:
+        extra_nix_config: experimental-features = nix-command flakes ca-derivations
+    - uses: cachix/cachix-action@v16
+      with:
+        name: ic-hs-test
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+    - name: nix-build
+      run: |
+        cachix watch-exec ic-hs-test -- \
+          nix build --max-jobs auto -L \
```

Comment:
Let's experiment with `--max-jobs auto`. This is run on every push to master. Not as critical as the release so we can experiment